src/hobbies.ts

```diff
@@ -0,0 +1,78 @@
+import * as http from 'http';
+
+const userHobbiesMap: { [userId: string]: string[] } = {};
+
+export const getUserHobbies = (req: http.IncomingMessage, res: http.ServerResponse) => {
+    const userId = getUserIdFromUrl(req.url!);
+    const hobbies = userHobbiesMap[userId] || [];
+
+    const response = {
+        data: {
+            hobbies,
+            links: {
+                self: `/api/users/${userId}/hobbies`,
+                user: `/api/users/${userId}`
+            }
+        },
+        error: null
+    };
+
+    res.writeHead(200, { 'Content-Type': 'application/json', 'Cache-Control': 'private, max-age=3600' });
+    res.end(JSON.stringify(response));
```

Comment:
https://medium.com/gist-for-js/use-of-res-json-vs-res-send-vs-res-end-in-express-b50688c0cddf